Nexus/Include/Nexus/Definitions/Money.hpp

```diff
@@ -28,16 +28,16 @@ namespace Details {
   };
 
   template<typename T>
-  constexpr const T MoneyDefinitions<T>::ZERO(0);
+  const T MoneyDefinitions<T>::ZERO(0);
```

Comment:
I think the Standard does not allow making anything `constexpr` out-of-line, anyway clang thinks so.
It compiles after removing `constexpr`, but let me know if you prefer a different solution.

Reply:
I think clang is wrong then, look into the issue and let me know. If clang is wrong then leave it as it was before, if clang is right then I'll have to think of a different way to implement this.

Reply:
I think I got it wrong, the issue only applies to CENT and BIP and is due to dependency on a non-constexpr:

```
[ 13%] Building CXX object Config/Compliance/CMakeFiles/ComplianceTests.dir/__/__/Source/ComplianceTests/CancelRestrictionPeriodComplianceRuleTester.cpp.o
In file included from /home/andrei/workspace/nexus/Nexus/Source/FeeHandlingTests/AsxtFeeHandlingTester.cpp:6:
In file included from /home/andrei/workspace/nexus/Nexus/Config/../Include/Nexus/FeeHandling/AsxtFeeTable.hpp:6:
/home/andrei/workspace/nexus/Nexus/Config/../Include/Nexus/Definitions/Money.hpp:37:42: error: constexpr variable 'CENT' must be initialized by a constant expression
  constexpr const T MoneyDefinitions<T>::CENT(ONE / 100);
                                         ^    ~~~~~~~~~
/home/andrei/workspace/nexus/Nexus/Config/../Include/Nexus/FeeHandling/AsxtFeeTable.hpp:91:50: note: in instantiation of static data member 'Nexus::Details::MoneyDefinitions<Nexus::Money>::CENT' requested here
    if(executionReport.m_lastPrice < 25 * Money::CENT) {
                                                 ^
/home/andrei/workspace/nexus/Nexus/Config/../Include/Nexus/Definitions/Money.hpp:37:47: note: read of non-constexpr variable 'ONE' is not allowed in a constant expression
  constexpr const T MoneyDefinitions<T>::CENT(ONE / 100);
                                              ^
/home/andrei/workspace/nexus/Nexus/Config/../Include/Nexus/Definitions/Money.hpp:37:47: note: in call to 'Money(ONE)'
/home/andrei/workspace/nexus/Nexus/Config/../Include/Nexus/Definitions/Money.hpp:21:20: note: declared here
    static const T ONE;
```
I'll try to figure out whether clang is right here.

Reply:
The minimal failing example is below:

```
template<typename T>
struct A {
  static const T s;
  static const T v;
 };
  
 template<typename T>
 constexpr const T A<T>::s = T(1);
 
template<typename T>
constexpr const T A<T>::v = A<T>::s;

int main(int, char**) {
  auto a = A<double>::v;
  return 0;
}
```

It compiles just fine with GCC but fails with clang 11. I could not find a reason for that, so I interpret it as a bug in clang and keep the original code, while applying patches to apply clang-tidy checks.